lib/administration/ticket/include/irods/ticket_administration.hpp

```diff
@@ -0,0 +1,578 @@
+        /// \param[in] admin_tag Struct tag to indicate admin privilige
+        /// \param[in] conn The communication object
+        /// \param[in] _type Specify the type of ticket wanting to be created
+        /// \param[in] obj_path The object path of the resource
+        /// \param[in] ticket_name The name of the ticket
+        ///
+        void create_ticket(admin_tag,
+                           RxComm& conn,
+                           type _type,
+                           std::string_view obj_path,
+                           std::string_view ticket_name);
+
+        ///
+        /// \brief Create a ticket object with a randomly generated ticket name using admin privilege
+        ///
+        /// \param[in] admin_tag Struct tag to indicate admin privilige
+        /// \param[in] conn The communication object
+        /// \param[in] _type Specify the type of ticket wanting to be created
+        /// \param[in] obj_path The object path of the resource
+        /// \return std::string The name of the ticket generated
+        ///
+        std::string create_ticket(admin_tag, RxComm& conn, type _type, std::string_view obj_path);
+
+        ///
+        /// \brief Delete the ticket that is specified
+        ///
+        /// \param[in] conn The communication object
+        /// \param[in] ticket_name Name of the ticket that is to be deleted
+        ///
+        void delete_ticket(RxComm& conn, std::string_view ticket_name);
+
+        ///
+        /// \brief Delete the ticket that is specified
+        ///
+        /// \param[in] conn The communication object
+        /// \param[in] ticket_id The ticket ID for the ticket that should be deleted
+        ///
+        void delete_ticket(RxComm& conn, int ticket_id);
+
+        ///
+        /// \brief Delete the ticket that is specified using admin privilige
+        ///
+        /// \param[in] admin_tag Struct tag to indicate admin privilige
+        /// \param[in] conn The communication Object
+        /// \param[in] ticket_name Name of the ticket that is to be deleted
+        ///
+        void delete_ticket(admin_tag, RxComm& conn, std::string_view ticket_name);
+
+        ///
+        /// \brief Delete the ticket that is specified using admin privilige
+        ///
+        /// \param[in] admin_tag Struct tag to indicate admin privilige
+        /// \param[in] conn The communication object
+        /// \param[in] ticket_id The ticket ID for the ticket that should be deleted
+        ///
+        void delete_ticket(admin_tag, RxComm& conn, int ticket_id);
+
+        ///
+        /// \brief Add the ticket constraint to the specified ticket
+        ///
+        /// \tparam ticket_constraint The template struct to indicate the type of constraint (only user, group, and host
+        /// constraint struct) \param[in] conn The communication object \param[in] ticket_name The name of the ticket
+        /// \param[in] constraint The constraint that should be added to the ticket
+        ///
```

Comment:
Add `/// \since 4.3.1` to each function's doxygen comments.

This line should be the last line in the doxygen comment block.